src/lib_ccx/matroska.c

```diff
@@ -738,14 +738,35 @@ void parse_segment(struct matroska_ctx* mkv_ctx)
     }
 }
 
+char *filename_without_ext(char* filename) {
+    char *returnname;
+    char *lastdot;
+    if (filename == NULL)
+         return filename;
+    if ((returnname = malloc (strlen (filename) + 1)) == NULL)
+        return filename;
+    strcpy (returnname, filename);
+    lastdot = strrchr (returnname, '.');
+    if (lastdot != NULL)
+        *lastdot = '\0';
+    return returnname;
+}
+
 char* generate_filename_from_track(struct matroska_ctx* mkv_ctx, struct matroska_sub_track* track)
-{
-    char* buf = malloc(sizeof(char) * 200);
+{ 
+  char* filename = malloc(sizeof(char)*(MAX_FILE_NAME_SIZE-30));
+  char * filename_non_ext;
+  filename[MAX_FILE_NAME_SIZE-31] = '\0';
+  strncpy(filename, mkv_ctx->filename,(MAX_FILE_NAME_SIZE-31));
+  filename_non_ext=filename_without_ext(filename);
```

Comment:
Also, why do you use additional variable `filename`? Why not passing mkv_ctx->filename to the function?